GVFS/FastFetch/Index.cs

```diff
@@ -187,15 +188,14 @@ private bool UpdateFileInformationFromWorkingTree(MemoryMappedViewAccessor index
                     {
                         foreach (FileInfo file in files)
                         {
-                            string gitPath = FromWindowsFullPathToGitRelativePath(file.FullName, this.repoRoot);
+                            string gitPath = FromDotnetFullPathToGitRelativePath(file.FullName, this.repoRoot);
                             long offset;
                             if (this.indexEntryOffsets.TryGetValue(gitPath, out offset))
                             {
-                                IndexEntry indexEntry = new IndexEntry(indexView, offset);
-                                indexEntry.Mtime = file.LastWriteTimeUtc;
-                                indexEntry.Ctime = file.CreationTimeUtc;
-                                indexEntry.Size = (uint)file.Length;
-                                Interlocked.Increment(ref updatedEntries);
+                                if (NativeMethods.StatAndUpdateIndexForFile(gitPath, indexView, offset))
+                                {
+                                    Interlocked.Increment(ref updatedEntries);
+                                }

```

Comment:
It looks like the old code did not have a failure mode once making it past `if (this.indexEntryOffsets.TryGetValue(gitPath, out offset))`.

If we get a failure now, should we log an error here?

Reply:
I'll log an error here. Note that before we incremented updatedEntries without checking for errors here, which seemed like a bug/mistake from looking at the internal repo's history.